Support blank character in Formula sql template

Hi. 
When I use `@Formula(sql = "...")`, I get an error if the `sql` contains blank characters: 
```
Caused by: org.babyfish.jimmer.meta.ModelException: Illegal property "xxx.totalThreePhaseVoltage", the formula sql template: Unexpected character ' ' the formula sql "%alias.voltage_phase_a + %alias.voltage_phase_b + %alias.voltage_phase_c"
	at app//org.babyfish.jimmer.meta.impl.ImmutablePropImpl.<init>(ImmutablePropImpl.java:256)
	at app//org.babyfish.jimmer.meta.impl.ImmutableTypeImpl$PropBuilder.build(ImmutableTypeImpl.java:1249)
	at app//org.babyfish.jimmer.meta.impl.ImmutableTypeImpl$BuilderImpl.build(ImmutableTypeImpl.java:1156)
	at app//org.faraday.rs.entity.device.CableDeviceReadingRecordDraft$$.<clinit>(CableDeviceReadingRecordDraft.kt:154)
	... 98 more
```

For example:

```kotlin
@Formula(sql = "%alias.voltage_phase_a + %alias.voltage_phase_b + %alias.voltage_phase_c")
val totalThreePhaseVoltage: BigDecimal
// Error: ModelException
```

When I change out the blank characters in it, the exception is gone.

```kotlin
@Formula(sql = "%alias.voltage_phase_a" +
        "+%alias.voltage_phase_b" +
        "+%alias.voltage_phase_c")
val totalThreePhaseVoltage: BigDecimal
// Is OK
```

--- a/project/jimmer-core/src/main/java/org/babyfish/jimmer/sql/meta/SqlTemplate.java
+++ b/project/jimmer-core/src/main/java/org/babyfish/jimmer/sql/meta/SqlTemplate.java
@@ -90,7 +90,6 @@ public abstract class SqlTemplate {
                     parenthesesDepth--;
                     break;
                 case ',':
-                case ' ':
                     if (parenthesesDepth == 0) {
                         throw new IllegalArgumentException(
                                 "Unexpected character '" +
@@ -100,6 +99,7 @@ public abstract class SqlTemplate {
                                         "\""
                         );
                     }
+                    break;
             }
             partBuilder.append(c);
         }

--- a/project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/classic/author/Author.kt
+++ b/project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/classic/author/Author.kt
@@ -37,6 +37,9 @@ interface Author {
     val fullName: String
         get() = "$firstName $lastName"
 
+    @Formula(sql = "length(%alias.FIRST_NAME) + length(%alias.LAST_NAME)")
+    val fullNameLength: Int
+
     /**
      * The fullName2 property
      */

--- a/project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/query/FetcherTest.kt
+++ b/project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/query/FetcherTest.kt
@@ -636,6 +636,27 @@ class FetcherTest : AbstractQueryTest() {
     }
 
     @Test
+    fun testSqlFormulaWithBlank() {
+        executeAndExpect(
+            sqlClient.createQuery(Author::class) {
+                where(table.firstName eq "Alex")
+                select(
+                    table.fetchBy {
+                        fullNameLength()
+                    }
+                )
+            }
+        ) {
+            statement(0).sql(
+                """select tb_1_.ID, length(tb_1_.FIRST_NAME) + length(tb_1_.LAST_NAME) 
+                    |from AUTHOR tb_1_ 
+                    |where tb_1_.FIRST_NAME = ?""".trimMargin()
+            )
+            rows("[{\"id\":2,\"fullNameLength\":9}]")
+        }
+    }
+
+    @Test
     fun testFetchIdView() {
         executeAndExpect(
             sqlClient

--- a/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/cache/MutateCacheTest.java
+++ b/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/cache/MutateCacheTest.java
@@ -226,6 +226,7 @@ public class MutateCacheTest extends AbstractQueryTest {
                     );
                     it.statement(2).sql(
                             "select tb_1_.ID, tb_1_.FIRST_NAME, tb_1_.LAST_NAME, tb_1_.GENDER, " +
+                                    "length(tb_1_.FIRST_NAME) + length(tb_1_.LAST_NAME), " +
                                     "concat(tb_1_.FIRST_NAME, ' ', tb_1_.LAST_NAME) " +
                                     "from AUTHOR tb_1_ " +
                                     "where tb_1_.ID in (?, ?, ?, ?, ?)"
@@ -351,6 +352,7 @@ public class MutateCacheTest extends AbstractQueryTest {
                     );
                     it.statement(2).sql(
                             "select tb_1_.ID, tb_1_.FIRST_NAME, tb_1_.LAST_NAME, tb_1_.GENDER, " +
+                                    "length(tb_1_.FIRST_NAME) + length(tb_1_.LAST_NAME), " +
                                     "concat(tb_1_.FIRST_NAME, ' ', tb_1_.LAST_NAME) " +
                                     "from AUTHOR tb_1_ " +
                                     "where tb_1_.ID in (?, ?, ?, ?, ?)"

--- a/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/fetcher/SimpleTest.java
+++ b/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/fetcher/SimpleTest.java
@@ -318,6 +318,36 @@ public class SimpleTest extends AbstractQueryTest {
     }
 
     @Test
+    public void testSqlFormulaWithBlank() {
+        AuthorTable table = AuthorTable.$;
+        executeAndExpect(
+                getSqlClient()
+                        .createQuery(table)
+                        .where(table.id().eq(eveId))
+                        .select(
+                                table.fetch(
+                                        AuthorFetcher.$.fullNameLength()
+                                )
+                        ),
+                ctx -> {
+                    ctx.sql(
+                            "select tb_1_.ID, length(tb_1_.FIRST_NAME) + length(tb_1_.LAST_NAME) " +
+                                    "from AUTHOR tb_1_ " +
+                                    "where tb_1_.ID = ?"
+                    );
+                    ctx.rows(
+                            "[" +
+                                    "--->{" +
+                                    "--->--->\"id\":\"fd6bb6cf-336d-416c-8005-1ae11a6694b5\"," +
+                                    "--->--->\"fullNameLength\":11" +
+                                    "--->}" +
+                                    "]"
+                    );
+                }
+        );
+    }
+
+    @Test
     public void testIdView() {
         BookTable table = BookTable.$;
         executeAndExpect(

--- a/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/loader/ManyToManyWithCacheTest.java
+++ b/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/loader/ManyToManyWithCacheTest.java
@@ -101,6 +101,7 @@ public class ManyToManyWithCacheTest extends AbstractCachedLoaderTest {
                     ).variables(learningGraphQLId3, graphQLInActionId3);
                     ctx.statement(1).sql(
                             "select tb_1_.ID, tb_1_.FIRST_NAME, tb_1_.LAST_NAME, tb_1_.GENDER, " +
+                                    "length(tb_1_.FIRST_NAME) + length(tb_1_.LAST_NAME), " +
                                     "concat(tb_1_.FIRST_NAME, ' ', tb_1_.LAST_NAME) " +
                                     "from AUTHOR tb_1_ " +
                                     "where tb_1_.ID in (?, ?, ?)"

--- a/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/Author.java
+++ b/project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/model/Author.java
@@ -29,6 +29,9 @@ public interface Author {
         return firstName() + ' ' + lastName();
     }
 
+    @Formula(sql = "length(%alias.FIRST_NAME) + length(%alias.LAST_NAME)")
+    int fullNameLength();
+
     @Formula(sql = "concat(%alias.FIRST_NAME, ' ', %alias.LAST_NAME)")
     String fullName2();
 

